SonarQube.Common/AnalysisProperties/AnalysisPropertyFileProvider.cs

```diff
@@ -34,7 +34,7 @@ public class AnalysisPropertyFileProvider : IAnalysisPropertyProvider
         /// <param name="commandLineArguments">List of command line arguments (optional)</param>
         /// <returns>False if errors occurred when constructing the provider, otherwise true</returns>
         /// <remarks>If a properties file could not be located then an empty provider will be returned</remarks>
-        public static bool TryCreateProvider(IEnumerable<ArgumentInstance> commandLineArguments, string defaultPropertiesFileDirectory, ILogger logger, out AnalysisPropertyFileProvider provider)
```

Comment:
Modified to return the interface rather than the concrete class. I've done the same for the file provider.